driver/src/main/java/com/consol/citrus/db/driver/JdbcPreparedStatement.java

```diff
@@ -330,19 +331,43 @@ public void setNClob(final int parameterIndex, final Reader reader) throws SQLEx
     }
 
     void setParameter(final int parameterIndex, final Object value){
-        if (parameters.size() > parameterIndex - 1) {
-            parameters.set(parameterIndex - 1,value);
-        }else {
-            parameters.add(parameterIndex - 1, value);
-        }
+        setParameter(String.valueOf(parameterIndex - 1), value);
```

Comment:
Yes because the mapping has grown this way. One could eliminate this but I think this is not worth the effort.

Reply:
It might cause unnecessary confusion when skimming through the code.
Concerning the scope of this PR it's not required to be updated, but I'd suggest to turn it into an improvement / cleanup issue.